channels/rdpdr/server/rdpdr_main.c

```diff
@@ -267,19 +273,31 @@ static int rdpdr_server_send_core_capability_request(RdpdrServerContext* context
 	header.PacketId = PAKID_CORE_SERVER_CAPABILITY;
 	numCapabilities = 5;
 	s = Stream_New(NULL, RDPDR_HEADER_LENGTH + 512);
+	if (!s)
+		return -1;
 	Stream_Write_UINT16(s, header.Component); /* Component (2 bytes) */
 	Stream_Write_UINT16(s, header.PacketId); /* PacketId (2 bytes) */
 	Stream_Write_UINT16(s, numCapabilities); /* numCapabilities (2 bytes) */
 	Stream_Write_UINT16(s, 0); /* Padding (2 bytes) */
-	rdpdr_server_write_general_capability_set(context, s);
-	rdpdr_server_write_printer_capability_set(context, s);
-	rdpdr_server_write_port_capability_set(context, s);
-	rdpdr_server_write_drive_capability_set(context, s);
-	rdpdr_server_write_smartcard_capability_set(context, s);
+	if (rdpdr_server_write_general_capability_set(context, s) != 0)
+		goto out_error;
+	if (rdpdr_server_write_printer_capability_set(context, s) != 0)
+		goto out_error;
+	if (rdpdr_server_write_port_capability_set(context, s) != 0)
+		goto out_error;
+	if (rdpdr_server_write_drive_capability_set(context, s) != 0)
+		goto out_error;
+	if (rdpdr_server_write_smartcard_capability_set(context, s) != 0)
+		goto out_error;
 	Stream_SealLength(s);
 	status = WTSVirtualChannelWrite(context->priv->ChannelHandle, (PCHAR) Stream_Buffer(s), Stream_Length(s), &written);
 	Stream_Free(s, TRUE);
-	return 0;
+	return status ? 0 : -1;
+
+out_error:
```

Comment:
suggest to move this label above line 294, delete line 298/299 and initialize status to FALSE